src/main/java/com/google/jenkins/plugins/k8sengine/VerificationTask.java

```
@@ -0,0 +1,134 @@
+    currentResult = KubernetesVerifiers.verify(kubectl, manifestObject);
+    if (isVerified()) {
+      consoleLogger.println(currentResult.toString());
+    }
+    return this;
+  }
+
+  /**
+   * @param kubectl KubectlWrapper object for issuing commands to kubernetes cluster
+   * @param manifestObjects list of manifest objects to verify
+   * @param consoleLogger PrintStream for outputting results (intended to be user facing)
+   * @param timeoutInMinutes Stop retrying verification after this many minutes
+   */
+  public static boolean verifyObjects(
+      @Nonnull KubectlWrapper kubectl,
+      @Nonnull List<Manifests.ManifestObject> manifestObjects,
+      @Nonnull PrintStream consoleLogger,
+      int timeoutInMinutes) {
+
+    List<VerificationTask> verificationTasks =
+        manifestObjects
+            .stream()
+            .map((manifestObject) -> new VerificationTask(kubectl, manifestObject, consoleLogger))
+            .collect(Collectors.toList());
+
+    Repeat.onlyIf(
+            (ctx) ->
+                !verificationTasks
+                    .stream()
+                    .map((task) -> task.isVerified()) // only repeat if we aren't all done
+                    .reduce(true, (acc, done) -> acc && done))
+        .backoff(Backoff.fixed(Duration.ofSeconds(VERIFICATION_DELAY)))
+        .timeout(Duration.ofMinutes(timeoutInMinutes))
+        // apply this repeat to the list of  VerificationTask's
+        .apply((Publisher<VerificationTask>) Flux.fromIterable(verificationTasks))
+        .filter(
+            (task) -> !task.isVerified()) // Don't try to verify objects that are already verified
+        .map((task) -> task.verify())
+        .subscribeOn(Schedulers.elastic()) // parallelize the verification
+        .doOnError(
+            (error) -> {
+              LOGGER.log(Level.SEVERE, "Unexpected error in verifyObjects()", error);
+              error.printStackTrace(consoleLogger); // report error
```

Should we consider throwing the exception so that downstream callers can decide what to do? Swallowing exceptions like this raises a red flag.

---

Nvm, just saw the error handling block below.